### DISCUSSION TITLE
Opensearch/Elasticsearch instance mock

### DISCUSSION THREAD

**Discussion by dacevedo12** [spulec/moto#8276] [General]
Is it within the scope of this project to provide a mock for the actual instances of those services? Like the index and search APIs

Could be achieved using something like https://pypi.org/project/openmock/ https://pypi.org/project/ElasticMock/

**Comment by bblommers** [comment]
Hi @dacevedo12, that is out of scope - Moto only focuses on the actual AWS API's. So you might have to monkey-patch this yourself to achieve this.

We have some info in our docs on how to patch boto3 services, so hopefully that is useful as an example/starting point on how to implement custom functionality around specific boto3 calls:

https://docs.getmoto.org/en/latest/docs/services/patching_other_services.html
